rokka/sourceimages.go

```diff
@@ -232,3 +232,64 @@ func (c *Client) DeleteDynamicMetadata(org, hash, name string, options DynamicMe
 	err = c.Call(req, &result, dynamicMetadataResponseHandler)
 	return result, err
 }
+
+// SetUserMetadata updates a source image by adding arbitrary metadata. If there were previous user metadata set on this source image,
+// they'll get overwritten.
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) SetUserMetadata(org, hash string, data io.Reader) error {
+	req, err := c.NewRequest(http.MethodPut, fmt.Sprintf("/sourceimages/%s/%s/meta/user", org, hash), data, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
+
+// UpdateUserMetadata updates a source image by adding arbitrary metadata. Previous values get merged with the new values.
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) UpdateUserMetadata(org, hash string, data io.Reader) error {
+	req, err := c.NewRequest(http.MethodPatch, fmt.Sprintf("/sourceimages/%s/%s/meta/user", org, hash), data, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
+
+// DeleteUserMetadata updates a source image by deleting existing metadata.
+//
+// See: https://rokka.io/documentation/references/image-metadata.html
+func (c *Client) DeleteUserMetadata(org, hash string) error {
+	req, err := c.NewRequest(http.MethodDelete, fmt.Sprintf("/sourceimages/%s/%s/meta/user", org, hash), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	return c.Call(req, nil, nil)
+}
```

Comment:
The SetUserMetadata, UpdateUserMetadata and DeleteUserMetadata functions look quite similar.
Maybe we should create a helper function which gets called by these 3 functions with the right parameters in order to reduce redundancy?

Reply:
fixed